google-cloud-storage/src/main/java/com/google/cloud/storage/GrpcStorageImpl.java

```diff
@@ -515,7 +493,11 @@ public Bucket update(BucketInfo bucketInfo, BucketTargetOption... options) {
 
   @Override
   public Blob update(BlobInfo blobInfo, BlobTargetOption... options) {
-    Opts<ObjectTargetOpt> opts = Opts.unwrap(options).resolveFrom(blobInfo).prepend(defaultOpts);
+    Opts<ObjectTargetOpt> unwrap = Opts.unwrap(options);
+    if (blobInfo.getModifiedFields().isEmpty()) {
+      return internalBlobGet(blobInfo.getBlobId(), unwrap.constrainTo(ObjectSourceOpt.class));
```

Comment:
so the behavior here is if the fields are not being modified we just perform a get and return the unmodified blob?

Reply:
Correct, while attempting to use those provided options that also can apply to Get